### PR TITLE
Remove NodeBuilder constructors with implicit NodeManager

### PR DESCRIPTION
**Pull request by daniel-larraz**
This change introduces some calls to `NodeManager::currentNM()`, which will be removed in a follow-up PR.

### CODE REVIEW

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/preprocessing/passes/bv_to_bool.cpp
Comment (daniel-larraz) [reply src/preprocessing/passes/bv_to_bool.cpp]: Fixed.

File: src/preprocessing/passes/foreign_theory_rewrite.cpp
Comment (daniel-larraz) [reply src/preprocessing/passes/foreign_theory_rewrite.cpp:124]: The issue is that `ForeignTheoryRewriter::reconstructNode` is a static method.

File: src/theory/arith/linear/arith_static_learner.cpp
Comment (daniel-larraz) [reply src/theory/arith/linear/arith_static_learner.cpp:145]: If this is the only change blocking this PR, I'd rather merge it now and apply the change after PR #11382 is merged.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**
